### PR TITLE
[#8215] Java: Use Java 8 map.putIfAbsent and map.computeIfAbsent

### LINKED ISSUE #8215: Java: Use Java 8 map.putIfAbsent and map.computeIfAbsent
Context: Someone wants to add a key-value pair to a map if the key-value pair does not exist.

Pre-Java 8:
```java
if (!map.containsKey(key)) {
    map.put(key, value);
}
```
Java 8:
```java
// If value is independent of key
map.putIfAbsent(key, value);

// If value is dependent of key
map.computeIfAbsent(key, key -> process(key));
```

For this one there are more than 20 places. Have to fix no less than 20 to qualify for a valid PR.

### PR DESCRIPTION
**Pull request by RonakLakhotia**
Fixes #8215

### CODE REVIEW

**Comment by RonakLakhotia** [comment]
Ready for review

**Comment by wkurniawan07** [comment]
We should fix the issue for good, actually. There are at least 3 more instances of this (hint: use IntelliJ to find), let's fix them as well.

**Comment by RonakLakhotia** [comment]
Found a couple more, not too sure of the third one.

File: src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java [hunk 1]
```diff
@@ -188,9 +188,7 @@ protected void markSectionAsSpecial(Map<String, Boolean> isSectionSpecialMapping
         // indicate that section group covers the section
         // and mark that this section is special
         String sectionGroupParamName = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionGroupIndex;
-        if (specialSectionsInSectionGroups.get(sectionGroupParamName) == null) {
-            specialSectionsInSectionGroups.put(sectionGroupParamName, new ArrayList<String>());
-        }
+        specialSectionsInSectionGroups.putIfAbsent(sectionGroupParamName, new ArrayList<String>());
         specialSectionsInSectionGroups.get(sectionGroupParamName).add(sectionToMark);
```
[the remark this answers is not in the document]
Comment (RonakLakhotia) [reply src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java]: you mean `specialSectionsInSectionGroups.putIfAbsent(sectionGroupParamName, new ArrayList<String>());` and
`specialSectionsInSectionGroups.get(sectionGroupParamName).add(sectionToMark);` can be combined?

**Comment by RonakLakhotia** [comment]
Ready for review

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 2]
```diff
@@ -1608,11 +1608,8 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserInSectionBy
                     frc.commentGiver = Const.DISPLAYED_NAME_FOR_ANONYMOUS_PARTICIPANT;
                 }
 
-                if (responseComments.get(frc.feedbackResponseId) == null) {
-                    responseComments.put(frc.feedbackResponseId,
-                            new ArrayList<FeedbackResponseCommentAttributes>());
-                }
-                responseComments.get(frc.feedbackResponseId).add(frc);
+                responseComments.computeIfAbsent(frc.feedbackResponseId, (String key) -> new ArrayList<>())
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: `key` does not need to be typed; there is type inference here.

File: src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java [hunk 3]
```diff
@@ -188,10 +188,8 @@ protected void markSectionAsSpecial(Map<String, Boolean> isSectionSpecialMapping
         // indicate that section group covers the section
         // and mark that this section is special
         String sectionGroupParamName = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionGroupIndex;
-        if (specialSectionsInSectionGroups.get(sectionGroupParamName) == null) {
-            specialSectionsInSectionGroups.put(sectionGroupParamName, new ArrayList<String>());
-        }
-        specialSectionsInSectionGroups.get(sectionGroupParamName).add(sectionToMark);
+        specialSectionsInSectionGroups.computeIfAbsent(sectionGroupParamName, (String key) -> new ArrayList<String>())
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java]: - `key` does not need to be typed; there is type inference here.
- `ArrayList<>()` (diamond constructor)

**Comment by RonakLakhotia** [comment]
Ready for review

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 4]
```diff
@@ -1608,11 +1608,8 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserInSectionBy
                     frc.commentGiver = Const.DISPLAYED_NAME_FOR_ANONYMOUS_PARTICIPANT;
                 }
 
-                if (responseComments.get(frc.feedbackResponseId) == null) {
-                    responseComments.put(frc.feedbackResponseId,
-                            new ArrayList<FeedbackResponseCommentAttributes>());
-                }
-                responseComments.get(frc.feedbackResponseId).add(frc);
+                responseComments.computeIfAbsent(frc.feedbackResponseId, String -> new ArrayList<>())
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: I'm surprised CheckStyle didn't flag this. `String` here is a lambda parameter and so it needs to follow camelCase, e.g. it should be `string`. Or if you want to be constant with you've done in another file, use `key`.

File: src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java [hunk 5]
```diff
@@ -188,10 +188,8 @@ protected void markSectionAsSpecial(Map<String, Boolean> isSectionSpecialMapping
         // indicate that section group covers the section
         // and mark that this section is special
         String sectionGroupParamName = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionGroupIndex;
-        if (specialSectionsInSectionGroups.get(sectionGroupParamName) == null) {
-            specialSectionsInSectionGroups.put(sectionGroupParamName, new ArrayList<String>());
-        }
-        specialSectionsInSectionGroups.get(sectionGroupParamName).add(sectionToMark);
+        specialSectionsInSectionGroups.computeIfAbsent(sectionGroupParamName, String -> new ArrayList<>())
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/controller/InstructorCourseInstructorAbstractAction.java]: Similarly here.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java (hunk 4 above)
Comment (RonakLakhotia) [reply src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: I will use key instead to maintain consistency.

**Comment by RonakLakhotia** [comment]
Ready for review

**Pull request merged**